docs/api-reference/core/resources/transform-feedback.md

```diff
@@ -1,135 +1,124 @@
-- `props.bindOnUse`=`true` - If true, binds and unbinds buffers before and after use, rather than right away when set. Workaround for a possible [Khronos/Chrome bug](https://github.com/KhronosGroup/WebGL/issues/2346).
+- `props.layout` = (`ShaderLayout`) - Layout of shader (for varyings)
+- `props.buffers` = (Object) - Map of location index or name to Buffer or BufferRange object. Used for varyings. If buffer parameters object is supplied, it contains following fields.
+  - `buffer` = (`Buffer`) - Buffer object to be bound.
+  - `byteOffset`= (Number, default: 0) - Byte offset for writes into the buffer.
+  - `byteLength`= (Number, default: buffer.byteLength) - Byte length, from offset, available for writes into the buffer.
 
 Notes:
 
 - `buffers` - will get bound to indices in the `GL.TRANSFORM_FEEDBACK_BUFFER` target.
 
-### delete() : TransformFeedback
+### destroy() : void
 
 Destroys a `TransformFeedback` object.
 
-WebGL APIS [`gl.deleteTransformFeedback`](https://developer.mozilla.org/en-US/docs/Web/API/WebGL2RenderingContext/deleteTransformFeedback)
+WebGL APIs [`gl.deleteTransformFeedback`](https://developer.mozilla.org/en-US/docs/Web/API/WebGL2RenderingContext/deleteTransformFeedback)
 
-### setBuffers(buffers: Object) : TransformFeedback
+### setBuffers(buffers: Object) : void
 
-- `buffers`=(Object) - Map of location index or name to Buffer object or buffer parameters object. If buffer parameters object is supplied, it contains following fields.
-  - `buffer`=(Buffer) - Buffer object to be bound.
-  - `byteOffset`=(Number, default: 0) - Byte offset that is used to start recording the data in the buffer.
-  - `byteSize`=(Number, default: remaining buffer size) - Size in bytes that is used for recording the data.
+- `buffers` = (Object) - Map of location index or name to Buffer or BufferRange object. Used for varyings. If buffer parameters object is supplied, it contains following fields.
+  - `buffer` = (Buffer) - Buffer object to be bound.
+  - `byteOffset` = (Number, default: 0) - Byte offset for writes into the buffer.
+  - `byteLength` = (Number, default: buffer.byteLength) - Byte length, from offset, available for writes into the buffer.
 
 Notes:
 
-- To use `gl.bindBufferRange`, either `offsetInByts` or `byteSize` must be specified, when only one is specified, default value is used for the other, when both not specified, `gl.bindBufferBase` is used for binding.
+- To use `gl.bindBufferRange`, `byteLength` must be specified. When not specified, `gl.bindBufferBase` is used for binding.
 
 WebGL APIs [`gl.bindBufferBase`](https://developer.mozilla.org/en-US/docs/Web/API/WebGL2RenderingContext/bindBufferBase), [`gl.bindBufferRange`](https://developer.mozilla.org/en-US/docs/Web/API/WebGL2RenderingContext/bindBufferRange)
 
-### begin(primitiveMode : GLEnum) : TransformFeedback
+### begin(topology : PrimitiveTopology = 'point-list') : void
```

Comment:
Good clarification, thanks! I'll go ahead and fix the parameter.

Reply:
I can't find these exact enum values (`'points' | 'lines' | 'triangles'`) elsewhere in the codebase, maybe the signature could be...

```typescript
begin(topology: 'point-list' | 'line-list' | 'triangle-list' = 'point-list'): void
```

... to use a subset of the WebGPU primitive topology enum.

Reply:
I think that from a semantic perspective, defining new constants to describe primitive type rather than how they are laid out makes most sense to me, but we can absolutely go with your proposal, this is rather obscure corner of the API so not worth much bikeshedding.

Reply:
Ok! I think this is feeling cleanest to me, but I'm totally fine with changing it too.